File: src/main/kotlin/dev/jefy/connectpro/engagement/applicaion/query/ReviewQuery.java
package dev.jefy.connectpro.engagement.applicaion.query;

import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

import java.util.List;

import dev.jefy.connectpro.engagement.applicaion.dtos.ReviewResponse;
import dev.jefy.connectpro.portfolio.domain.vo.ServiceId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface ReviewQuery {
    @Nullable ReviewResponse findServiceReviewForCurrentUser(ServiceId serviceId);
    List<ReviewResponse> findByService(ServiceId serviceId);
}

File: src/main/kotlin/dev/jefy/connectpro/engagement/application/query/ReviewQuery.kt
package dev.jefy.connectpro.engagement.application.query

import dev.jefy.connectpro.engagement.application.dtos.ReviewResponse
import dev.jefy.connectpro.portfolio.domain.vo.ServiceId


interface ReviewQuery {
    fun findServiceReviewForCurrentUser(serviceId: ServiceId): ReviewResponse
    fun findByService(serviceId: ServiceId): List<ReviewResponse>
}